### ISSUE TITLE
[SDC] Add help button view more feature when there's long text

### ISSUE THREAD

**Issue by shelaghm** [google/android-fhir#1970]
**Problem**
The help box does not support a `see more` feature if there's long text. Showing a see more button is helpful when there's a lot of text to display, but the developer doesn't want to show it all upon first opening the help box, as this can be distracting. 

**Describe the solution you'd like**
Add a way to show a short preview of text (3-4 lines) and a view more button to see all the content.

**2 options**
`Option 1` - Add a `see more` / `see less` button to expand/hide text inside the help box. 

- Can still see questionnaire which can either be good (seeing question context) or bad (distracting)
- Questionnaire options would be displayed below help box
- More information dense compared to a dialog

`Option 2` - Open a modal dialog when tapping `see more`. This dialog would need to be scrollable. 

- Don't see the questionnaire which can either be good (not distracting) or bad (not enough context)
- Would need to enable scrolling inside the dialog box. 
- Dialog option takes up more space, so less information dense

Screenshots of options below. Input and feedback is welcome on which direction you think would be most helpful for your users?

<img width="949" alt="Screenshot 2023-04-12 at 15 29 16" src="https://user-images.githubusercontent.com/86378921/231600252-db4a2edb-9ac2-47c5-8fd5-fe4afbf4835f.png">

cc @santosh-pingle @kunjan8794 @khyativyasargus @jingtang10

**Comment by santosh-pingle** [comment]
@shelaghm 
Option 2 looks good to me, but my only concern is that the user has to jump around to different places just to access information.

**Comment by santosh-pingle** [comment]
@aditya-07 @nsabale7 @joiskash @FikriMilano 
Maybe you would also like to add your opinion here?

**Comment by santosh-pingle** [comment]
@shelaghm 
We need to decide on options before starting the implementation.

**Comment by jingtang10** [comment]
@shelaghm is there an option where we can just use the modal when the user clicks the ? the first time. seems like the cleanest approach.

**Comment by joiskash** [comment]
> @aditya-07 @nsabale7 @joiskash @FikriMilano Maybe you would also like to add your opinion here?

Hmm I dont really have a strong opinion here. Both seem to work fine

**Comment by shelaghm** [comment]
@jingtang10 yes, that is an option too. I'm curious to hear what @rowo @henryrae think? What do you think of additional information being shown only in a dialog modal compared to the information being shown in-line in the questionnaire? related to https://github.com/google/android-fhir/issues/1689

See below for the 3 options we are discussing
<img width="477" alt="Screenshot 2023-04-20 at 17 05 08" src="https://user-images.githubusercontent.com/86378921/233511226-654f014a-8224-45a8-bc17-aadc63da8297.png">

**Comment by santosh-pingle** [comment]
I am of the opinion that option 3 would be the best choice.

**Comment by dubdabasoduba** [comment]
@shelaghm My sense of UX is not the best but could we combine options 1 and 3? 

Have a max amount of characters that allow us to show the help text inline like in option 1. We however don't give the show more options. 

If the max amount of characters is surpassed we then default to showing the modal in option 3. 

This way we don't always show the modal for really short help texts. I am not sure if that makes sense. 

@rowo @HenryRae what do you think?

**Comment by ppanchal-argus** [comment]
@shelaghm Thanks for the options, looks really good!

We had a discussion with the Em care team and **Option 2** seems more in line with the requirements.
**Reasons for option 2:** There are usually scenario where the caregiver might only need the short description in the help to carry on with the tasks and only open view more modal when more information is required. 

Option 3 kind of forces user to open a modal each time they want to look at the help.

Also had a short discussion from a different perspective on including both option 1 & 2. Depending on the severity of information included in help it could either be a modal (when its more urgent) or be like option 1 (when its more normal info).

**Comment by jingtang10** [comment]
the other problem with this expandable approach is that it's hard to decide where to cut the text short - i think we might end up with more variabilities that we'll have to deal with and that'll increase complexity in both the api and the implementations.

for short text, can we just use a different field - help is for additional context. if something should be displayed all the time, use `instructions`?

**Comment by jingtang10** [comment]
> @shelaghm My sense of UX is not the best but could we combine options 1 and 3?
> 
> Have a max amount of characters that allow us to show the help text inline like in option 1. We however don't give the show more options.
> 
> If the max amount of characters is surpassed we then default to showing the modal in option 3.
> 
> This way we don't always show the modal for really short help texts. I am not sure if that makes sense.
> 
> @rowo @HenryRae what do you think?

then we'll have to define the max amount of characters... can the questionnaire author just decide what goes in the short instruction?

**Comment by HenryRae** [comment]
I think option 3 is more idea,
My thinking is that user might see tis form more often and will be seeing similar info over time for i.e patients with same condition/careplan. I think not showing the more info would be best solution with `?` is clicked.

The other alternative would be option 2 with character limit like @dubdabasoduba mentioned, abit tweaked though. i.e remove help title, move description box right below the question since it's related to the question and not the answers
Played around with your screenshot @shelaghm. Let me know what you think of this
<img width="411" alt="Screenshot 2023-05-10 at 16 16 12" src="https://github.com/google/android-fhir/assets/31955469/8275a66c-c1c5-436d-a6e8-781b1d90e8ec">

**Comment by jingtang10** [comment]
Thanks @HenryRae for the update!

@parthfloyd @kunjan8794 i think we really need to think more carefully about the cut off point if we go with option 2... henry just mentioned maybe 3 lines, but i'm sure we'll get more requirements to do 2 lines or 4 lines - i really think it will not look good with different question texts, especially once we author more questionnaires.

**Comment by ppanchal-argus** [comment]
Thanks @HenryRae & @jingtang10 . Yes, the cutoff point would indeed be tricky to go with a fixed value.  We'll be having discussion on this again & I'll share the team's feedback latest by Friday PM.

**Comment by santosh-pingle** [comment]
@ppanchal-argus 
Do you have any feedback you would like to share?

**Comment by khyativyasargus** [comment]
@santosh-pingle @jingtang10 Apologies we don't have any final conclusion on this yet. Maybe we can set up a call with @jingtang10 and Emcare team to discuss all the scenarios and get to a conclusion. Will that work?

**Comment by santosh-pingle** [comment]
@khyativyasargus 
We currently hold a weekly UX design call, during which we discuss UX-related queries with @shelaghm  and @jingtang10. The call takes place on Wednesdays at 9:30 pm IST. I'd like to inquire if this timing suits your availability, or if you would prefer to suggest an alternative time.

**Comment by FikriMilano** [comment]
fyi @rowo

**Comment by rowo** [comment]
Thanks @FikriMilano. This sure is tricky @shelaghm

First, I'm assuming we're using something like the [Wordpress <more> tag](https://wordpress.com/support/wordpress-editor/blocks/more-block/) approach where we leave the cutoff to the user. So if the user wants the second screen on option 1 to show up (without the show less button), that's up to them.

But this thread is about not bombarding the user with all that information. So starting with the question, "is a preview blurb ever necessary?" — The answer seems like yes in https://github.com/google/android-fhir/issues/1970#issuecomment-1529684879. That eliminates option 3 (plus I agree I don't want to open that every time — and even if we gave the option to either use the help box or modal, the inconsistency of what happens when you press that question mark wouldn't be great). 

So @HenryRae's [modified option 2](https://github.com/google/android-fhir/issues/1970#issuecomment-1542199403), seems nice in that it offers a different interaction to open the modal vs. the help icon (which means the user can know what happens next). 

That being said, I think option 1 would cover most use cases in help content length. I'm looking at the nytimes mobile app Russia-Ukraine War story updates here and if they want more content than a few screen lengths, seems they are expecting too much. Plus, the way they say "urgent cases" in https://github.com/google/android-fhir/issues/1970#issuecomment-1529684879 to me says the modal approach is perceived as trying to solve a different problem than help content. I'm curious what "urgent help" is versus "normal help".

If we could build a custom interaction in option 1 — an unassuming-until-you-need-it way to have the "show less" button accessible all of the time— that would be even better. 

I do currently have a separate, related use case in the "expecting too much" realm. They wanted an almost  reference amount of content accessible — like a glossary/dictionary of 50 terms or a toolkit of 10+ methods to do something — so it needed links/navigation, which the modal would lend itself to.
